test/e2e/accounts/test-snap-accounts.spec.js

```diff
@@ -0,0 +1,452 @@
+  // (in Jest we could do this with test.each, but that does not exist here)
+  [
+    ['#personalSign', 'sync'],
+    ['#personalSign', 'approve'],
+    ['#personalSign', 'reject'],
+    ['#signTypedData', 'sync'],
+    ['#signTypedData', 'approve'],
+    ['#signTypedData', 'reject'],
+    ['#signTypedDataV3', 'sync'],
+    ['#signTypedDataV3', 'approve'],
+    ['#signTypedDataV3', 'reject'],
+    ['#signTypedDataV4', 'sync'],
+    ['#signTypedDataV4', 'approve'],
+    ['#signTypedDataV4', 'reject'],
+    ['#signPermit', 'sync'],
+    ['#signPermit', 'approve'],
+    ['#signPermit', 'reject'],
+  ].forEach(([locatorID, flowType]) => {
+    // generate title of the test from the locatorID and flowType
+    let title = `can ${locatorID} (${
+      flowType === 'sync' ? 'sync' : 'async'
+    } flow`;
+
+    title += flowType === 'sync' ? ')' : ` ${flowType})`;
+
+    it(title, async function () {
+      await withFixtures(
+        accountSnapFixtures(this.test.title),
+        async ({ driver }) => {
+          const isAsyncFlow = flowType !== 'sync';
+
+          await installSnapSimpleKeyring(driver, isAsyncFlow);
+
+          const newPublicKey = await makeNewAccountAndSwitch(driver);
+
+          await openDapp(driver);
+
+          await signData(driver, locatorID, newPublicKey, flowType);
+        },
+      );
+    });
+  });
+
+  it('can connect to the Test Dapp, then #signTypedDataV3, disconnect then connect, then #signTypedDataV4 (async flow approve)', async function () {
+    await withFixtures(
+      {
+        dapp: true,
+        fixtures: new FixtureBuilder().build(),
+        ganacheOptions,
+        failOnConsoleError: false,
+        title: this.test.title,
+      },
+      async ({ driver }) => {
+        const flowType = 'approve';
+        const isAsyncFlow = flowType !== 'sync';
+
+        await installSnapSimpleKeyring(driver, isAsyncFlow);
+
+        const newPublicKey = await makeNewAccountAndSwitch(driver);
+
+        // open the Test Dapp and connect Account 2 to it
+        await connectAccountToTestDapp(driver);
+
+        // do #signTypedDataV3
+        await signData(driver, '#signTypedDataV3', newPublicKey, flowType);
+
+        // disconnect from the Test Dapp
+        await disconnectFromTestDapp(driver);
+
+        // reconnect Account 2 to the Test Dapp
+        await connectAccountToTestDapp(driver);
+
+        // do #signTypedDataV4
+        await signData(driver, '#signTypedDataV4', newPublicKey, flowType);
+      },
+    );
+  });
+
+  /**
+   * @param {Driver} driver
+   * @param {string} flowType
+   */
+  async function importPrivateKeyAndTransfer1ETH(driver, flowType) {
+    const isAsyncFlow = flowType !== 'sync';
+
+    await installSnapSimpleKeyring(driver, isAsyncFlow);
+    await importKeyAndSwitch(driver);
+
+    // send 1 ETH from Account 2 to Account 1
+    await sendTransaction(driver, PUBLIC_KEY, 1, isAsyncFlow);
+
+    if (isAsyncFlow) {
```

Comment:
NIT: If statements are generally discouraged in e2e due to adding complexity and variability: https://docs.google.com/document/d/1eQyeScSZQlzGLapef_8PANkU_ObNPJcuMCP040T2gfM/edit#heading=h.njqh4zz4ol4w